### PR TITLE
HHKB YDKB Yang BT Controller

### PR DESCRIPTION
**Pull request by kanru**
## Description

Adds support for the Yang controller for HHKB Pro 2.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* The previous PR #11011 was closed and I can't reopen it so I'm opening a new one.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hhkb/yang/rules.mk [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# MCU frequency
+F_CPU = 8000000
+
+# Bootloader selection
+BOOTLOADER = lufa-ms
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/hhkb/yang/rules.mk:13]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/hhkb/yang/config.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/hhkb/yang/config.h:124]:
```suggestion
```

File: keyboards/hhkb/yang/keymaps/kanru/keymap.c [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+#define BASE 0
+#define HHKB 1
+#define MOUSE 2
```
Comment (drashna) [inline keyboards/hhkb/yang/keymaps/kanru/keymap.c:24]:
```suggestion
enum custom_layers {
    BASE,
    HHKB,
    MOUSE,
};
```

File: keyboards/hhkb/yang/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define BASE 0
+#define HHKB 1
```
Comment (drashna) [inline keyboards/hhkb/yang/keymaps/default/keymap.c:22]:
```suggestion
enum custom_layers {
    BASE,
    HHKB,
};
```

File: keyboards/hhkb/yang/rules.mk [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/hhkb/yang/rules.mk:18]:
```suggestion
```

File: keyboards/hhkb/yang/rules.mk [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+
+LAYOUTS = 60_hhkb
+
+# Disable bluetooth until the UART code is merged
+# BLUETOOTH = AdafruitBLE_UART
```
Comment (drashna) [inline keyboards/hhkb/yang/rules.mk:24]: This is no longer the syntax used.  You need to set the driver, *and* enable it. 
For here, just setting the driver is fine: 
```suggestion
BLUETOOTH_DRIVER = BluefruitLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**